http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -429,14 +429,22 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
 
   override def listAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, params: Map[String, String])(implicit
     as: Ask[F, AppContext]
-  ): F[Vector[ListAppResponse]] = for {
-    paramMap <- F.fromEither(processListParameters(params))
-    allClusters <- KubernetesServiceDbQueries
-      .listFullAppsByWorkspaceId(Some(workspaceId), paramMap._1, paramMap._2)
-      .transaction
+  ): F[Vector[ListAppResponse]] =
+    for {
+      // Make sure that the user still has access to the resource parent workspace
+      hasWorkspacePermission <- authProvider.isUserWorkspaceReader(
+        WorkspaceResourceSamResourceId(workspaceId),
+        userInfo
+      )
+      _ <- F.raiseUnless(hasWorkspacePermission)(ForbiddenError(userInfo.userEmail))
```

Comment:
I was considering whether 403 or 404 makes more sense for no-workspace-permission (e.g. we may want to pretend the workspace doesn't exist). But I am not sure, 403 seems fine I think.

Reply:
From a security perspective, 404 would make more sense I think, but we are going to have users lose access to their resources, and getting a 403 will let them know exactly why